### DISCUSSION TITLE
blog/total-guide-to-jest-esm-and-angular

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#11] [General]
# blog/total-guide-to-jest-esm-and-angular

Let's learn how to speed up our Angular Jest tests by more than 100% by switching to Jest ESM (a notoriously problematic migration) and how to solve all the troublesome issues that tend to pop up along the way!

https://angularexperts.io/blog/total-guide-to-jest-esm-and-angular

**Comment by timsar2** [comment]
From angular 16, ngcc was removed. Should we still using jest-preset-angular or there is a better way?

**Reply by sherlock1982** [reply]
You should still be using it as it provides a boilerplate. If you use jest without Angular CLI you will have to call `setupZoneTestEnv` from your `test.ts`. Otherwise don't call it

**Comment by eniomarku** [comment]
I am using angular version 16, which by default have under @angular/cdk only fesm2022. Should down-leveling to fesm2015 would work out of the box If there is no fesm2015 under my @angular/cdk directory?

**Comment by bhalperin** [comment]
Would highly appreciate an update to this blog, relevant to Angular 18 and Nx.
All my attempts to fix the dreaded message **SyntaxError: Cannot use import statement outside a module** have failed. I'm totally confused....
I'm using Angular 18 inside an Nx monorepo.

**Reply by raboud** [reply]
been fighting this and with angualar 18 and Nx for awhile....have you made any head way

**Reply by bhalperin** [reply]
Unfortunately not 😞

**Comment by sherlock1982** [comment]
It is worth mentioning the downsides of this migration. They're well described here `https://jestjs.io/docs/ecmascript-modules`. Specifically:
1. `jest.mock` won't work and you will have to mock via DI. That is bad if you highly rely on it
2. you will have to import from `@jest/globals`. This is even good as brings some extra type checking

Overall I don't fully understand why you mention the speed increase. You have to process all your ESM to CJS and by default you're using cache. So you process only once and there should be no noticable performance increase.

**Comment by sherlock1982** [comment]
From my experience - I tried it, fixed all jest.mock, interops e.t.c. Unfortunately I see that the memory of nodejs keeps growing until it crashes with OOM. 
I don't see it if i run the same tests but with cjs
